### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by hawkrives**

### CODE REVIEW

**Review by drewvolz — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['javascript', 'python', 'java', 'cpp']
```
Comment (rye) [inline .github/workflows/codeql-analysis.yml:27]: We don't have `cpp` files, or at least a build system that works to do so.  Do we need to keep that in the build matrix?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file. 
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (rye) [inline .github/workflows/codeql-analysis.yml:57]: It looks like the Java build doesn't work; do we need to `yarn install --frozen-lockfile` to get the .gradle files?

**Comment by hawkrives** [comment]
@rye both excellent questions!

**Comment by hawkrives** [comment]
Do we want this? I'm tempted to close it. I don't even remember what this is supposed to do?

**Comment by drewvolz** [comment]
> Discover vulnerabilities across a codebase with CodeQL, our industry-leading semantic code analysis engine. CodeQL lets you query code as though it were data. Write a query to find all variants of a vulnerability, eradicating it forever.

Looks like we thought it had potential in helping us write safer code / find vulnerabilities.

https://codeql.github.com/

**Comment by hawkrives** [comment]
I'm going to go ahead and close this - it looks really powerful, but … eh. Maybe too powerful for what we have here? I don't mind if someone else takes it on, though!

**Pull request closed without merging**